frame/contracts/src/wasm/runtime.rs

```diff
@@ -1859,4 +1870,43 @@ define_env!(Env, <E: Ext>,
 			Err(_) => Ok(ReturnCode::CallRuntimeReturnedError),
 		}
 	},
+
+	// Computes the ECDSA public key on the given hash of message and signature.
+	//
+	// Returns the result directly into the given output buffer.
+	//
+	// # Parameters
+	//
+	// - `signature_ptr`: the pointer into the linear memory where the signature
+	//					  is placed. Should be decodable as a 65 bytes. Traps otherwise.
+	// - `message_hash_ptr`: the pointer into the linear memory where the message
+	// 						 hash is placed. Should be decodable as a 32 bytes. Traps otherwise.
+	// - `output_ptr`: the pointer into the linear memory where the output
+	//                 data is placed. The function will write the result
+	//                 directly into this buffer.
+	// # Errors
+	//
+	// `ReturnCode::RecoveryFailed`
+	[__unstable__] seal_ecdsa_recovery(ctx, signature_ptr: u32, message_hash_ptr: u32, output_ptr: u32) -> ReturnCode => {
+		ctx.charge_gas(RuntimeCosts::EcdsaRecovery)?;
+
+		let mut signature: [u8; 65] = [0; 65];
+		ctx.read_sandbox_memory_into_buf(signature_ptr, &mut signature)?;
+		let mut message_hash: [u8; 32] = [0; 32];
+		ctx.read_sandbox_memory_into_buf(message_hash_ptr, &mut message_hash)?;
+
+		let result = secp256k1_ecdsa_recover_compressed(&signature, &message_hash);
```

Comment:
We shouldn't to this call here directly. Functions in runtime should:
* Charge weight
* Decode params
* Call into business logic residing in an  `crate::exec::Ext` implementation (e.G `ctx.ext.ecdsa_recover`)
* Encode result

Reason for this indirection through `Ext` is so that we can add tests in `exec.rs` for the business logic and tests for the wasm integration in `wasm/mod.rs`. You should add this indirection and add those  tests. The tests in `test.rs` are end-to-end tests (the one you added) which can be added on-top or are required to test more complicated functions.

It is a bit boilerplate but specifically the tests in `exec.rs` allow us to properly test the business logic without needing to hand write wasm code.

You can look into the `seal_hash_*` functions which are quite similar to this function. They add another in-between function for de-duplication which is not needed here, though.